### DISCUSSION TITLE
Support limit URL parameter on REST API ListView results

### DISCUSSION THREAD

**Discussion by john-terraspace** [jsforce/jsforce#1431] [Ideas]
Returning the results from the ListView method currently only returns 25 records. This is a Salesforce default, however it can be overridden.  See: https://trailhead.salesforce.com/trailblazer-community/feed/0D54V00007T4HFrSAN

Made a quick test modifying the sobject.js file and this works:

Calling the method with a limit:
const results = ((await conn.sobject('Lead').listview(listViewId).results(()=>{}, 2000)) as any);

**Comment by cristiand391** [comment]
the `limit` param is documented so it's seem safe to add:
https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_listviewresults.htm

This should be a non-breaking change in jsforce v3 (`3.0` branch), `listview` would take an optional 2nd arg as an object where you specify the limit and offset here:
https://github.com/jsforce/jsforce/blob/573b869aff8a4ad01d0969c64c1590383326cc6d/src/sobject.ts#L472

PRs welcome.
